src/agent/communicator/src/http_client.cpp

```diff
@@ -140,7 +140,7 @@ namespace http_client
 
             boost::system::error_code ec;
 
-            co_await socket->AsyncConnect(results, ec);
+            co_await socket->AsyncConnect(results, ec, std::chrono::seconds(TIMEOUT_DEFAULT)); // NOLINT
```

Comment:
We should use `// NOLINT(name-of-warning-being-suppresed)` so we know what was the issue and we can keep an eye for it.

Reply:
Done